contracts/abstracts/TokenCollector.sol

```diff
@@ -61,8 +66,11 @@ abstract contract TokenCollector {
     ) private {
         require(amount < uint256(type(uint160).max), "TokenCollector: permit2 amount too large");
         if (data.length > 0) {
-            (bool success, ) = uniswapPermit2.call(abi.encodePacked(IUniswapPermit2.permit.selector, data));
-            require(success, "TokenCollector: permit2 permit failed");
+            (address owner, IUniswapPermit2.PermitSingle memory permit, bytes memory permitSig) = abi.decode(
+                data,
+                (address, IUniswapPermit2.PermitSingle, bytes)
+            );
+            IUniswapPermit2(uniswapPermit2).permit(owner, permit, permitSig);
```

Comment:
Permit2 uses custom error. I didn't find a good way to keep the original error along the way to our client from the `result` returned by `call`, so I switched to explicitly call the `permit` function

```solidity
(bool success, bytes memory result) = uniswapPermit2.call(abi.encodePacked(IUniswapPermit2.permit.selector, data));
if (!success) {
    // how to revert?
}
```

Reply:
I think we can refactor this in the later starge, let's focus on the functionality first.